fiftyone/core/metadata.py

```diff
@@ -417,6 +475,92 @@ def compute_metadata(
             raise ValueError(msg)
 
 
+def _bulk_metadata_writer(
+    collection, update_queue, stop_signal, batch_size=1000
+):
+    ops = []
+    while True:
+        item = update_queue.get()
+        if item is stop_signal:
+            break
+
+        ops.append(item)
+        if len(ops) >= batch_size:
+            collection.bulk_write(ops)
+            ops.clear()
+    if ops:
+        collection.bulk_write(ops)
+        ops.clear()
+
+
+def _compute_metadata_map_fcn(args):
+    oid, filepath, media_type, cache = args
+    metadata = _get_metadata_from_cache_or_build_from_src(
+        filepath, media_type, cache=cache
+    )
+    if metadata:
+        return UpdateOne(
+            {"_id": oid}, {"$set": {"metadata": metadata.to_mongo()}}
+        )
+
+
+def _compute_metadata_iter_fcn(sample_collection):
+    cache = {}
+
+    for sid, filepath, media_type in sample_collection._iter_values(
+        ["_id", "filepath", "_media_type"], _allow_missing=True
+    ):
+        yield sid, tuple([sid, filepath, media_type, cache])
+
+
+def _compute_metadata_single(
```

Comment:
If we use `SampleCollection.map_samples()` then none of this code would be necessary because passing `num_workers<=1` to it automatically uses a non-thread/process pool implementation of mapping.

More reasons to use the core abstractions that we build!

Reply:
you're right. I missed it in the mapper.py